content/blog/2018/egress-performance/index.md

```diff
@@ -0,0 +1,122 @@
+
+        - -x
+        - "169.47.232.211/32"
+
+{{< image width="70%" ratio="60%"
+    link="./case1_sidecar_bypass3.png"
+    caption="Traffic to external MongoDB by-passing the sidecar"
+    >}}
+
+### Case 2: Through the sidecar, with service entry
+
+This is the default configuration when the sidecar is injected into the application pod. All messages are intercepted by the sidecar and routed to the destination according to the configured rules, including the communication with external services. The MongoDB was defined as a `ServiceEntry`.
+
+{{< image width="70%" ratio="60%"
+    link="./case2_sidecar_passthru3.png"
+    caption="Sidecar intercepting traffic to external MongoDB"
+    >}}
+
+### Case 3: Egress gateway
+
+The egress gateway and corresponding destination rule and virtual service resources are defined for accessing MongoDB. All traffic to and from the external DB goes through the egress gateway (envoy).
+
+{{< image width="70%" ratio="60%"
+    link="./case3_egressgw3.png"
+    caption="Introduction of the egress gateway to access MongoDB"
+    >}}
+
+### Case 4: Mutual TLS between sidecars and the egress gateway
+
+In this case, there is an extra layer of security between the sidecars and the gateway, so some impact in performance is expected.
+
+{{< image width="70%" ratio="60%"
+    link="./case4_egressgw_mtls3.png"
+    caption="Enabling mTLS between sidecars and the egress gateway"
+    >}}
+
+### Case 5: Egress gateway with SNI proxy
+
+This scenario ia used to evaluate the case where another proxy is required to access wildcarded domains. This may be required due current limitations of envoy. An nginx proxy was created as sidecar in the egress gateway pod.
+
+{{< image width="70%" ratio="60%"
+    link="./case5_egressgw_sni_proxy3.png"
+    caption="Egress gateway with additional SNI Proxy"
+    >}}
+
+## Environment
+
+* Istio version: 1.0.2
+* K8s version: `1.10.5_1517`
+* Acmeair App: 4 services (1 replica of each), inter-services transactions, external Mongo DB, avg payload: 620 bytes.
+
+## Results
+
+`Jmeter` was used to generate the workload which consisted in a sequence of 5-minute runs, each one using a growing number of clients making http requests. The number of clients used were 1, 5, 10, 20, 30, 40, 50 and 60.
+
+### Throughput
+
+The chart below shows the throughput obtained for the different cases:
+
+{{< image width="75%" ratio="60%"
+    link="./throughput3.png"
+    caption="Throughput obtained for the different cases"
+    >}}
+
+As we can see, there is no major impact in having sidecars and the egress gateway between the application and the external MongoDB, but enabling mTLS and then adding the SNI proxy caused a degradation in the throughput of about 10% and 24%, respectively.
```

Comment:
we -> you

Reply:
mTLS -> mutual TLS
(here and anywhere else)

Reply:
ok, fixed the typo and the new comments.
how do I rebase? 
Can it stay in the 2018 folder? It was supposed to be published in December, butdue vacations and other issues, I could not go back to it. The build it was based on is kind of old now.
I don't need use ratio for images, but is it okay to keep it now?